DataFormats/Portable/interface/PortableHostCollection.h

```diff
@@ -47,27 +50,86 @@ class PortableHostCollection {
   PortableHostCollection& operator=(PortableHostCollection const&) = delete;
 
   // movable
-  PortableHostCollection(PortableHostCollection&&) = default;
-  PortableHostCollection& operator=(PortableHostCollection&&) = default;
+  PortableHostCollection(PortableHostCollection&& other):
+    buffer_(std::move(other.buffer_)),
+    layout_(std::move(other.layout_)),
+    view_(std::move(other.view_))
+          {
+            other.buffer_.reset();
+            other.view_ = View();
+          }
+  PortableHostCollection& operator=(PortableHostCollection&& other) {
+    // Protection for self-assignment without if-statements, following
+    // what was suggested in the C++ core guidelines
+    {
+      auto tmp = std::move(other.buffer_);
+      other.buffer_.reset();
+      buffer_.reset();
```

Comment:
if this `reset()` necessary ?

Reply:
Maybe ?

Reply:
> Is this `reset()` necessary ?

I'd think (now) it would not be necessary, and I was just following the pattern in an example https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-move-self (that uses raw pointers where the `delete` in the assigned-to-object's member is necessary). I'd expect the assignment on the following line
```cpp
buffer_ = std::move(tmp);
```
to work equally well in both cases of self-assignment and assignment from a different object. Any existing value in `buffer_` (be it a valid Buffer or a moved-from Buffer) should get destructed first before initializing the content of `buffer_` from `tmp`.